packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -296,6 +335,18 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
       this._connections = new ec2.Connections({});
     }
 
+    if (!cdk.Token.isUnresolved(props.artifactS3EncryptionMode) &&
```

Comment:
I refactored the creation of artifactConfig into createArtifactConfig.
However, since kmsKey needs to be set as a property, I made it so that checking and setting are done here.

Please provide your opinion if there is a better way to do this.​​​​​​​​​​​​​​​​

Reply:
I like the approach, just a couple of adjustments:
* Why not move all validations inside the method?
* No need to declare the `public readonly encryptionKey?: kms.IKey;` variable
* Watch out for the typo in the function name
Suggestion:
```ts
  private createArtifactConfig(props: CanaryProps): CfnCanary.ArtifactConfigProperty | undefined {
    if (!props.artifactS3EncryptionMode) {
      return undefined;
    }

    const isNodeRuntime = !cdk.Token.isUnresolved(props.runtime) && props.runtime.family === RuntimeFamily.NODEJS;
    const isArtifactS3EncryptionModeDefined = !cdk.Token.isUnresolved(props.artifactS3EncryptionMode) && props.artifactS3EncryptionMode;
    const isArtifactS3KmsKeyDefined = !cdk.Token.isUnresolved(props.artifactS3KmsKey) && props.artifactS3KmsKey;

    if (isArtifactS3EncryptionModeDefined &&
      props.artifactS3EncryptionMode !== ArtifactsEncryptionMode.KMS &&
      isArtifactS3KmsKeyDefined
    ) {
      throw new Error('A customer-managed KMS key was provided, but the encryption mode is not set to SSE-KMS.');
    }

    // Only check runtime family is nodejs because versions prior to syn-nodejs-puppeteer-3.3 are deprecated and can no longer be configured.
    if (!isNodeRuntime && isArtifactS3EncryptionModeDefined) {
      throw new Error(`Artifact encryption is only supported for canaries that use Synthetics runtime version syn-nodejs-puppeteer-3.3 or later, got ${props.runtime.name}.`);
    }

    let encryptionKey: kms.IKey | undefined;
    if (props.artifactS3EncryptionMode === ArtifactsEncryptionMode.KMS) {
      // Kms Key is set or generated for using `createArtifactConfig`
      encryptionKey = props.artifactS3KmsKey ?? new kms.Key(this, 'Key', { description: `Created by ${this.node.path}` });
    }

    encryptionKey?.grantEncryptDecrypt(this.role);

    return {
      s3Encryption: {
        encryptionMode: props.artifactS3EncryptionMode,
        artifactS3KmsKeyArn: encryptionKey?.keyArn,
      },
    };
  }
```